util/oidc/provider.go

```diff
@@ -107,6 +107,13 @@ func (p *providerImpl) Verify(tokenString string, argoSettings *settings.ArgoCDS
 		// Token must be verified for at least one allowed audience
 		for _, aud := range allowedAudiences {
 			idToken, err = p.verify(aud, tokenString, false)
+			tokenExpiredError := &gooidc.TokenExpiredError{}
+			if errors.As(err, &tokenExpiredError) {
```

Comment:
in think you could just do:
```golang
if errors.Is(err, gooidc.TokenExpiredError) {
```
and avoid having to declare this additional var: `tokenExpiredError`

Reply:
Weirdly enough, it requires an instantiated object rather than a type. 🤷‍♂️

Reply:
Oh! Remembered why we don't do that.

The value of `gooidc.TokenExpiredError` is dynamic:

```go
func (e *TokenExpiredError) Error() string {
	return fmt.Sprintf("oidc: token is expired (Token Expiry: %v)", e.Expiry)
}
```

That type also does not implement `Is`. So `&TokenExpiredError{} != &TokenExpiredError{}` for any two errors where the expiry is not the same.

From docs for `errors.Is`:

```go
// An error is considered to match a target if it is equal to that target or if
// it implements a method Is(error) bool such that Is(target) returns true.
```

`errors.As` works because the concrete type of `TokenExpiredError` is assignable to a `TokenExpiredError`. From the docs for `errors.As`:

```go
// An error matches target if the error's concrete value is assignable to the value
// pointed to by target, or if the error has a method As(interface{}) bool such that
// As(target) returns true. In the latter case, the As method is responsible for
// setting target.
```